docs/v2_proposal.rst

```diff
@@ -0,0 +1,39 @@
+=======================
+OCX-Library V2 Proposal
+=======================
+
+The intention of this library is to act as a wrapper around the underlying client libraries.
+A user will import all of the "python-[service]client" libraries they would like to use, in addition to ours. 
+Our library relies on the existance of the other client libraries in the python environment.
+
+Code control flow:
+------------------
+* First call
+    - initialize x keystone clients
+    - initialize x service clients, one for each keystone client
+    - call the function with kwargs on the x service clients
+* Every subsequent call, until clients time out
+    - retrieve the service clients stored in the user's running software 
+    - call the function with kwargs on the x service clients
+* Once clients time out, we inform the user, and ask if they would like to re-initialize the clients
```

Comment:
Currently timeout range is 2 hours I believe, which is a fairly long time. Do we need to extend that? How? Why?